core/src/main/java/org/apache/carbondata/core/util/CleanFilesUtil.java

```diff
@@ -206,10 +206,12 @@ private static void getStaleSegmentFiles(CarbonTable carbonTable, List<String> s
   /**
    * This method will delete all the empty partition folders starting from the table path
    */
-  private static void deleteEmptyPartitionFoldersRecursively(CarbonFile tablePath) {
+  public static void deleteEmptyPartitionFoldersRecursively(CarbonFile tablePath) {
     CarbonFile[] listOfFiles = tablePath.listFiles();
     if (listOfFiles.length == 0) {
       tablePath.delete();
+      // if parent file folder also empty then delete that too.
+      deleteEmptyPartitionFoldersRecursively(tablePath.getParentFile());
```

Comment:
this may delete the database also, right? why add this here?

Reply:
this i added as part of bug fix. If partition folder is empty after moving to trash, it will delete empty folder. After moving if database folder also empty then it will delete that too.